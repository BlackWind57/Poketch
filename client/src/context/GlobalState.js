import React, { createContext, useReducer } from 'react';

import appReducer from './AppReducer';

const initialState = {
    pokemons: [],
    loading: true,
    error: ''
};

export const GlobalContext = createContext(initialState);

export const GlobalProvider = ({ children }) => {
  const [state, dispatch] = useReducer(appReducer, initialState);

  function addPokemonToMyList(pokemon) {
    dispatch({
      type: "ADD_POKEMON",
      payload: pokemon
    });
  }

  function removePokemonFromMyList(pokemon) {
    dispatch({
      type: "REMOVE_POKEMON",
      payload: pokemon
    });
  }

  function fetchingMyPokemons ( data ) {
    dispatch ( {
      type: 'FETCH_SUCCESS',
      payload: data
    });
  }

  function fetchingError ( error ) {
    dispatch ( {
      type: 'FETCH_ERROR',
      payload: error
    });
  }

  function fetchData() {
    try {
      if (state.loading) {
        fetch("/.netlify/functions/server/api/my-list")
        .then ( response => response.json() )
        .then ( data => {
          fetchingMyPokemons ( data );
        })
        .catch ( error => {
          console.error( error );
          fetchingError ( error );
        });
      }

    } catch (e) {
      console.log(e);
    }
  }

  return (
    <GlobalContext.Provider
      value={{
        pokemons: state.pokemons,
        loading: state.loading,
        error: state.error,
        addPokemonToMyList,
        removePokemonFromMyList,
        fetchData,
        fetchingMyPokemons,
        fetchingError
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};
