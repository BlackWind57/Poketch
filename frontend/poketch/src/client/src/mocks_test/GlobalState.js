import React, { createContext, useReducer } from 'react';

import appReducer from '../context/AppReducer';

const initialState = {
    pokemons: [],
    loading: true,
    error: ''
};

export const GlobalContext = createContext(initialState);

export const GlobalProvider = ({ children }) => {
  const [state, dispatch] = useReducer(appReducer, initialState);

  function addPokemonToMyList(pokemon) {
    dispatch({
      type: "ADD_POKEMON",
      payload: pokemon
    });
  }

  function removePokemonFromMyList(pokemon) {
    dispatch({
      type: "REMOVE_POKEMON",
      payload: pokemon
    });
  }

  function fetchingMyPokemons ( data ) {
    dispatch ( {
      type: 'FETCH_SUCCESS',
      payload: data
    });
  }

  function fetchingError ( error ) {
    dispatch ( {
      type: 'FETCH_ERROR',
      payload: error
    });
  }

  function fetchData ( success ) {
    try {
      if (state.loading) {
        if ( success ) {
          // using a sample data from the graphql
          fetchingMyPokemons ( [
            {
              _id: "6068a9c0a9393a000841ff58",
              name: 'squirtle',
              nick_name: 'Turtle',
              image_url: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png'
            },
            {
              _id: "6069cc7406a087000713e81e",
              name: 'wartortle',
              nick_name: 'negi',
              image_url: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/8.png'
            },
            {
              _id: "6069d5290ccc4300098e0b99",
              name: 'wartortle',
              nick_name: 'dasd',
              image_url: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/8.png'
            }
          ]);
        }
        else {
          fetchingError ( "You have an error" );
        }
      }
    } catch (e) {
      console.log(e);
    }
  }

  return (
    <GlobalContext.Provider
      value={{
        pokemons: state.pokemons,
        loading: state.loading,
        error: state.error,
        addPokemonToMyList,
        removePokemonFromMyList,
        fetchData,
        fetchingMyPokemons,
        fetchingError
      }}
    >
      {children}
    </GlobalContext.Provider>
  );
};
