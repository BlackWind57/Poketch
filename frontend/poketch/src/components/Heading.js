import React from "react";
import { Link } from "react-router-dom";

export const Heading = () => {
  return (
    <div>
      <div className="flex items-center mt-24 mb-10">
        <div className="flex-grow text-right px-4 py-2 m-2">
          <Link to="/my-list">
            <button className="bg-green-400 hover:bg-green-500 text-white font-semibold py-2 px-4 rounded inline-flex items-center">
              <span className="pl-2">My Pokemons</span>
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
};
