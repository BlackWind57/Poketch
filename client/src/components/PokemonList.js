import React, { useEffect, useState, useContext } from 'react';
import { gql, useQuery } from '@apollo/client';
import { Link } from 'react-router-dom';
import { GlobalContext } from '../context/GlobalState';


import styled from '@emotion/styled';

import String from '../lib/String';

// Query to fetch the list of pokemons from Graphql
const GET_POKEMONS = gql `
  query pokemons($limit: Int, $offset: Int) {
    pokemons(limit: $limit, offset: $offset) {
      count
      next
      previous
      status
      message
      results {
        url
        name
        image
      }
    }
  }
`;

// CSS START
const Item = styled.div`
  text-align: center;
`;

const Image = styled.img `
  margin: 0 auto;
`;

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  margin-left: 20px;
  margin-right: 20px;
  justify-content: center;
  padding-bottom: 10px;

  /* Responsive layout - makes a one column layout instead of a two-column layout */
  @media (max-width: 320) {
    flex-direction: column;
  }
`;

const Box = styled.div`
  padding-left: 5px;
  padding-right: 5px;
  margin-left: 5px;
  margin-right: 5px;
  height: 20vh;
  -webkit-align-items: center;
  align-items: center;
  --tw-bg-opacity: 1;
  background-color: rgba(243, 244, 246, var(--tw-bg-opacity));
  margin-bottom: 10px;
  --tw-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(0 0 #0000, 0 0 #0000), var(--tw-shadow);
`;

const ShowMoreLink = styled.a`
  padding-bottom: 20px;
  text-decoration: underline;
  font-size: 14px;
`;

// Main Component of Pokemon List
export const PokemonList = () => {

  const { fetchData } = useContext ( GlobalContext );

  // set initial state of showMore for number of pokemons
  const [showMore, setShowMore] = useState ({
    count: 9
  });

  // Fetch pokemons from Pokemon Graphql as data
  const { loading, error, data } = useQuery(GET_POKEMONS, {
    variables: { limit: showMore.count }
  });

  useEffect ( () => {
      // Fetch pokemons from db
      fetchData();
  });

  if (loading) return 'Loading...';
  if (error) return `Error! ${error.message}`;

  // show more pokemons in increments of 9 items
  const handleShowMore = ( count ) => {
    setShowMore ( { ...showMore, count: count + 9 } );
  }

  // Render the Component
  return (
    <>
      <React.Fragment>
        {data.pokemons.results.length > 0 ? (
          <>
            <Container data-testid="test">
              <React.Fragment>
                {data.pokemons.results.map( (pokemon, i) => (
                  <Link
                    to={`/detail/${pokemon.name}`}
                    title=""
                    key={pokemon.name}
                  >
                    <Box>
                        <Item>
                          <Image src={pokemon.image} alt={ pokemon.url } />
                          <p className="text-gray-900 leading-none">
                            { String.capitalizeFirstLetter( pokemon.name ) }
                          </p>
                        </Item>
                    </Box>
                  </Link>
                ))}
              </React.Fragment>
            </Container>
            <React.Fragment>
              { (
                  <ShowMoreLink
                    onClick={ (e) => handleShowMore( showMore.count ) }>
                    Show More
                  </ShowMoreLink>
                )
              }
            </React.Fragment>
          </>
        ) : (
          <p className="text-center bg-gray-100 text-gray-500 py-5" data-testid="test">No data.</p>
        )}
      </React.Fragment>
    </>
  );
};
