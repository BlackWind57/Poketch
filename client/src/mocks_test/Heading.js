import React, { useContext } from "react";
import { Link } from "react-router-dom";
import { GlobalContext } from './GlobalState';

import styled from '@emotion/styled';

const HeadingDiv = styled.div `
  display: flex;
  align-items: center;
  margin-top: 96px;
  margin-bottom: 40px;

  div {
    flex-grow: 1;
    text-align: right;
    padding: 8px 16px;
    margin: 8px;
  }

  button {
    background-color: rgba(52, 211, 153, 1);
    color: white;
    font-weight: 600;
    padding: 8px 16px;
    border-radius: 4px;
    display: inline-flex;
    place-items: center;
  }

  button:hover {
    background-color: rgba(16, 185, 129, 1);
  }
`;

export const Heading = () => {
  const { pokemons } = useContext(GlobalContext);

  return (
    <>
      <HeadingDiv>
        <div>
          <Link to="/my-list">
            <button>
              <span id="pokemon-title">My Pokemons ({ pokemons.length })</span>
            </button>
          </Link>
        </div>
      </HeadingDiv>
    </>
  );
};
