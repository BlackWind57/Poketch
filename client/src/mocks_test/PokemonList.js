import React, { useEffect, useContext } from 'react';
import { Link } from 'react-router-dom';
import { GlobalContext } from './GlobalState';


import styled from '@emotion/styled';

import String from '../lib/String';

// Sample query result from Graphql
const pokemons = [
  {
    "url": "https://pokeapi.co/api/v2/pokemon/2/",
    "name": "ivysaur",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/3/",
    "name": "venusaur",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/4/",
    "name": "charmander",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/5/",
    "name": "charmeleon",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/6/",
    "name": "charizard",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/7/",
    "name": "squirtle",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/8/",
    "name": "wartortle",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/8.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/9/",
    "name": "blastoise",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/9.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/10/",
    "name": "caterpie",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/10.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/11/",
    "name": "metapod",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/11.png"
  },
  {
    "url": "https://pokeapi.co/api/v2/pokemon/12/",
    "name": "butterfree",
    "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/12.png"
  }
];

/////////////////// CSS START
const Item = styled.div`
  text-align: center;
`;

const Image = styled.img `
  margin: 0 auto;
`;

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  margin-left: 20px;
  margin-right: 20px;
  justify-content: center;

  /* Responsive layout - makes a one column layout instead of a two-column layout */
  @media (max-width: 320) {
    flex-direction: column;
  }
`;

const Box = styled.div`
  padding-left: 5px;
  padding-right: 5px;
  margin-left: 5px;
  margin-right: 5px;
  height: 20vh;
  -webkit-align-items: center;
  align-items: center;
  --tw-bg-opacity: 1;
  background-color: rgba(243, 244, 246, var(--tw-bg-opacity));
  margin-bottom: 10px;
  --tw-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(0 0 #0000, 0 0 #0000), var(--tw-shadow);
`;
//////////////////// CSS ENDS



export const PokemonList = () => {
  const { fetchData } = useContext ( GlobalContext );

  // Fetch pokemons from Pokemon Graphql as data

  useEffect ( () => {
      // Fetch pokemons from db
      fetchData( true );
  });

  return (
    <>
        {pokemons.length > 0 ? (
          <Container data-testid="pokemon-list">
            <React.Fragment>
              {pokemons.map( (pokemon, i) => (
                <Link
                  to={`/detail/${pokemon.name}`}
                  title={pokemon.name}
                  key={pokemon.name}
                  id={pokemon.name}
                >
                  <Box>
                      <Item>
                        <Image src={pokemon.image} alt={ pokemon.url } />
                        <p className="text-gray-900 leading-none">
                          { String.capitalizeFirstLetter( pokemon.name ) }
                        </p>
                      </Item>
                  </Box>
                </Link>
              ))}
            </React.Fragment>
          </Container>
        ) : (
          <p className="text-center bg-gray-100 text-gray-500 py-5">No data.</p>
        )}
    </>
  );
};
