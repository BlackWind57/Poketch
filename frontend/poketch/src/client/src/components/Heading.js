import React, { useContext } from "react";
import { Link } from "react-router-dom";
import { GlobalContext } from '../context/GlobalState';

export const Heading = () => {
  const { pokemons } = useContext(GlobalContext);

  return (
    <div>
      <div className="flex items-center mt-24 mb-10">
        <div className="flex-grow text-right px-4 py-2 m-2">
          <Link to="/my-list">
            <button className="bg-green-400 hover:bg-green-500 text-white font-semibold py-2 px-4 rounded inline-flex items-center">
              <span className="pl-2">My Pokemons ({ pokemons.length })</span>
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
};
