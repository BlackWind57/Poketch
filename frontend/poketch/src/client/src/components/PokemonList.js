import React, { useEffect, useContext } from 'react';
import { gql, useQuery } from '@apollo/client';
import { Link } from 'react-router-dom';
import { GlobalContext } from '../context/GlobalState';


import styled from '@emotion/styled';

import String from '../lib/String';

const GET_POKEMONS = gql `
  query pokemons($limit: Int, $offset: Int) {
    pokemons(limit: $limit, offset: $offset) {
      count
      next
      previous
      status
      message
      results {
        url
        name
        image
      }
    }
  }
`;

const gqlVariables = {
  limit: 100,
  offset: 0,
};

/////////////////// CSS START
const Item = styled.div`
  text-align: center;
`;

const Image = styled.img `
  margin: 0 auto;
`;

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  margin-left: 20px;
  margin-right: 20px;
  justify-content: center;

  /* Responsive layout - makes a one column layout instead of a two-column layout */
  @media (max-width: 320) {
    flex-direction: column;
  }
`;

const Box = styled.div`
  padding-left: 5px;
  padding-right: 5px;
  margin-left: 5px;
  margin-right: 5px;
  height: 20vh;
  -webkit-align-items: center;
  align-items: center;
  --tw-bg-opacity: 1;
  background-color: rgba(243, 244, 246, var(--tw-bg-opacity));
  margin-bottom: 10px;
  --tw-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.1), 0 1px 2px 0 rgba(0, 0, 0, 0.06);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000), var(0 0 #0000, 0 0 #0000), var(--tw-shadow);
`;
//////////////////// CSS ENDS



export const PokemonList = () => {
  const { fetchData } = useContext ( GlobalContext );

  // Fetch pokemons from Pokemon Graphql as data
  const { loading, error, data } = useQuery(GET_POKEMONS, {
    variables: gqlVariables,
  });

  useEffect ( () => {
      // Fetch pokemons from db
      fetchData();
  });

  if (loading) return 'Loading...';
  if (error) return `Error! ${error.message}`;

  console.log('Response from server', data);

  return (
    <React.Fragment>
      {data.pokemons.results.length > 0 ? (
        <Container data-testid="test">
          <React.Fragment>
            {data.pokemons.results.map( (pokemon, i) => (
              <Link
                to={`/detail/${pokemon.name}`}
                title=""
                key={pokemon.name}
              >
                <Box>
                    <Item>
                      <Image src={pokemon.image} alt={ pokemon.url } />
                      <p className="text-gray-900 leading-none">
                        { String.capitalizeFirstLetter( pokemon.name ) }
                      </p>
                    </Item>
                </Box>
              </Link>
            ))}
          </React.Fragment>
        </Container>
      ) : (
        <p className="text-center bg-gray-100 text-gray-500 py-5" data-testid="test">No data.</p>
      )}
    </React.Fragment>
  );
};
