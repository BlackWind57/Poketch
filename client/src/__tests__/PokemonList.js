import React from 'react';
import { HashRouter } from 'react-router-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { GlobalContext, GlobalProvider } from '../mocks_test/GlobalState';
import { PokemonList } from '../mocks_test/PokemonList';
import { Heading } from '../mocks_test/Heading';
import TestRenderer from 'react-test-renderer';

const pokemons = [ 'ivysaur', 'venusaur',
'charmander', 'charmeleon', 'charizard',
'squirtle', 'wartortle', 'blastoise', 'caterpie',
'metapod', 'butterfree' ];


function generatePokemonList() {
  return new TestRenderer.create(
    <HashRouter>
       <GlobalContext.Provider value={{
           pokemons: [],
           loading: false,
           error: "",
           fetchData: jest.fn()
            }} >
              <PokemonList />
       </GlobalContext.Provider>
     </HashRouter>
    );
}

function generateHeadingAndPokemonList() {
  return new TestRenderer.create (
    <HashRouter>
       <GlobalContext.Provider value={{
           pokemons: [
             {
               _id: "020394",
               nick_name: "Test nick name",
               name: "squirtle"
             }
           ],
           loading: false,
           error: "",
           fetchData: jest.fn()
            }} >
              <Heading />
              <PokemonList />
       </GlobalContext.Provider>
     </HashRouter>
  );
}

afterEach(cleanup);

describe ("Test Pokemon list", () => {
  it("snapshots testing", () => {
    let root;
    TestRenderer.act(() => {
      root = generatePokemonList();
    });

    // make assertions on root
    expect(root.toJSON()).toMatchSnapshot();
  });

  it("renders correctly based on sample data", () => {

    const element = generatePokemonList();

      expect ( element.toJSON().children.length ).toBe (pokemons.length);
      for (var i = 0; i < pokemons.length; i++) {
        expect(element.toJSON().children[i].props.title).toBe(pokemons[i]);
      }
  });
});

describe ("Test Pokemon list with Heading", () => {
  it ("renders correctly on load", () => {
    let element;
    TestRenderer.act(() => {
      element = generateHeadingAndPokemonList();
    });

    // make assertions on root
    expect(element.toJSON()).toMatchSnapshot();
  });

  it('Tests my pokemons initial count should be 0', () => {
    let element;
    TestRenderer.act(() => {
      element = new TestRenderer.create (
        <HashRouter>
           <GlobalContext.Provider value={{
               pokemons: [],
               loading: false,
               error: "",
               fetchData: jest.fn()
                }} >
                  <Heading />
                  <PokemonList />
           </GlobalContext.Provider>
         </HashRouter>
      );
    });

    let title = element.root.findByProps ({id: 'pokemon-title'});

    expect ( title.children ).toEqual ([ 'My Pokemons (', '0', ')' ]);
  });

  it('Tests my pokemons count should exist 1 pokemon', function () {
    let element;
    TestRenderer.act(() => {
      element = generateHeadingAndPokemonList();
    });

    let title = element.root.findByProps ({id: 'pokemon-title'});

    expect ( title.children ).toEqual ([ 'My Pokemons (', '1', ')' ]);
  });

});


/*
describe ( "Going to pokemon details page", () => {
  it("goes to each individual pokemons correctly", () => {
    const element = generatePokemonList();

    console.log(element.root.findByProps({to:'/detail/squirtle'}) );

    let link = element.root.findByProps({id:'squirtle'});
    TestRenderer.act (() => {
      fireEvent.click ( link );
    });

    console.log(element.toJSON());
    expect (element.root.findByType('Title')).toBeTruthy();
   });
});
*/
