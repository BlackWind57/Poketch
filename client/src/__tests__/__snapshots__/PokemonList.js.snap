// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Pokemon list snapshots testing 1`] = `
<div
  className="css-wcex5t"
  data-testid="pokemon-list"
>
  <a
    href="#/detail/ivysaur"
    id="ivysaur"
    onClick={[Function]}
    title="ivysaur"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/2/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Ivysaur
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/venusaur"
    id="venusaur"
    onClick={[Function]}
    title="venusaur"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/3/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Venusaur
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/charmander"
    id="charmander"
    onClick={[Function]}
    title="charmander"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/4/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Charmander
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/charmeleon"
    id="charmeleon"
    onClick={[Function]}
    title="charmeleon"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/5/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Charmeleon
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/charizard"
    id="charizard"
    onClick={[Function]}
    title="charizard"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/6/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Charizard
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/squirtle"
    id="squirtle"
    onClick={[Function]}
    title="squirtle"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/7/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Squirtle
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/wartortle"
    id="wartortle"
    onClick={[Function]}
    title="wartortle"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/8/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/8.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Wartortle
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/blastoise"
    id="blastoise"
    onClick={[Function]}
    title="blastoise"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/9/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/9.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Blastoise
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/caterpie"
    id="caterpie"
    onClick={[Function]}
    title="caterpie"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/10/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/10.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Caterpie
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/metapod"
    id="metapod"
    onClick={[Function]}
    title="metapod"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/11/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/11.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Metapod
        </p>
      </div>
    </div>
  </a>
  <a
    href="#/detail/butterfree"
    id="butterfree"
    onClick={[Function]}
    title="butterfree"
  >
    <div
      className="css-1282ew1"
    >
      <div
        className="css-1yltisf"
      >
        <img
          alt="https://pokeapi.co/api/v2/pokemon/12/"
          className="css-12k99am"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/12.png"
        />
        <p
          className="text-gray-900 leading-none"
        >
          Butterfree
        </p>
      </div>
    </div>
  </a>
</div>
`;

exports[`Test Pokemon list with Heading renders correctly on load 1`] = `
Array [
  <div
    className="css-1gwvebz"
  >
    <div>
      <a
        href="#/my-list"
        onClick={[Function]}
      >
        <button>
          <span
            id="pokemon-title"
          >
            My Pokemons (
            1
            )
          </span>
        </button>
      </a>
    </div>
  </div>,
  <div
    className="css-wcex5t"
    data-testid="pokemon-list"
  >
    <a
      href="#/detail/ivysaur"
      id="ivysaur"
      onClick={[Function]}
      title="ivysaur"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/2/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Ivysaur
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/venusaur"
      id="venusaur"
      onClick={[Function]}
      title="venusaur"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/3/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Venusaur
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/charmander"
      id="charmander"
      onClick={[Function]}
      title="charmander"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/4/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Charmander
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/charmeleon"
      id="charmeleon"
      onClick={[Function]}
      title="charmeleon"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/5/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Charmeleon
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/charizard"
      id="charizard"
      onClick={[Function]}
      title="charizard"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/6/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Charizard
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/squirtle"
      id="squirtle"
      onClick={[Function]}
      title="squirtle"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/7/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Squirtle
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/wartortle"
      id="wartortle"
      onClick={[Function]}
      title="wartortle"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/8/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/8.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Wartortle
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/blastoise"
      id="blastoise"
      onClick={[Function]}
      title="blastoise"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/9/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/9.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Blastoise
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/caterpie"
      id="caterpie"
      onClick={[Function]}
      title="caterpie"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/10/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/10.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Caterpie
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/metapod"
      id="metapod"
      onClick={[Function]}
      title="metapod"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/11/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/11.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Metapod
          </p>
        </div>
      </div>
    </a>
    <a
      href="#/detail/butterfree"
      id="butterfree"
      onClick={[Function]}
      title="butterfree"
    >
      <div
        className="css-1282ew1"
      >
        <div
          className="css-1yltisf"
        >
          <img
            alt="https://pokeapi.co/api/v2/pokemon/12/"
            className="css-12k99am"
            src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/12.png"
          />
          <p
            className="text-gray-900 leading-none"
          >
            Butterfree
          </p>
        </div>
      </div>
    </a>
  </div>,
]
`;
